curvefs/src/metaserver/metastore.cpp

```diff
@@ -453,6 +453,8 @@ MetaStatusCode MetaStoreImpl::CreateInode(const CreateInodeRequest* request,
     param.type = request->type();
     param.parent = request->parent();
     param.rdev = request->rdev();
+    param.tv_sec = request->createsec();
+    param.tv_nsec = request->creatensec();
```

Comment:
Consider backward compatibility, even though is a bug, because we cannot upgrade all the components at once.

For most cases, we upgrade server-side components first, then the client-side.

So, in this case, when we upgrade metaserver to a newer version, these two fields are missing if the older client sends create inode requests which mean `tv_sec` and `tv_nsec` will be 0, then we end up with another bug.

A proper way is to test whether the request has `createsec` and `creatensec`, if it has, then we use these two timestamps, otherwise, we fallback to previous buggy version.

And for `InodeParam`, we can use `absl::optional` to wrap a timespec, like
```cpp
struct InodeParam {
   // other fields
   absl::optional<timespec> timestamp;
};
```

And in here, we can do this
```cpp
if (request->has_createsec() && request->has_creatensec()) {
    param.timestamp.emplace(request->createsec(), request->creatensec());
}
```

Reply:
> Consider backward compatibility, even though is a bug, because we cannot upgrade all the components at once.
> 
> For most cases, we upgrade server-side components first, then the client-side.
> 
> So, in this case, when we upgrade metaserver to a newer version, these two fields are missing if the older client sends create inode requests which mean `tv_sec` and `tv_nsec` will be 0, then we end up with another bug.
> 
> A proper way is to test whether the request has `createsec` and `creatensec`, if it has, then we use these two timestamps, otherwise, we fallback to previous buggy version.
> 
> And for `InodeParam`, we can use `absl::optional` to wrap a timespec, like
> 
> ```c++
> struct InodeParam {
>    // other fields
>    absl::optional<timespec> timestamp;
> };
> ```
> 
> And in here, we can do this
> 
> ```c++
> if (request->has_createsec() && request->has_creatensec()) {
>     param.timestamp.emplace(request->createsec(), request->creatensec());
> }
> ```

fix.